src/hooks/useDetectNetworkConnection.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  useCallback,
+  createContext,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { useNetInfo } from '@react-native-community/netinfo';
+import { t } from 'i18next';
+import Toast, { ToastProps } from 'react-native-root-toast';
+
+const DetectNetworkConnectionContext = createContext<{
+  isConnected: boolean | null;
+  checkConnection?: () => void;
+}>({
+  isConnected: true,
+});
+
+export const DetectNetworkConnectionProvider = ({
+  children,
+  toastOverrides,
+}: {
+  children?: React.ReactNode;
+  toastOverrides?: ToastProps;
+}) => {
+  const { isConnected } = useNetInfo();
+  const [forceCheck, setForceCheck] = useState<boolean>(false);
+  const toastRef = useRef(undefined);
+  const showToast = useCallback(
+    () =>
+      Toast.show(t('network-unstable', 'Your network is unstable.'), {
```

Comment:
Do we want to more accurately say something like, "Your Internet connection is unstable"?

This is a minor though, okay to keep it as is.  Probably just me who gets pedantic thinking, "what?  my network is perfectly stable,  it's my connection to the Internet that is a problem!" 😄

Reply:
I suppose the message is stating that your LAN is down. Most users will not make that distinction between network and internet however there still may be some! I will update it in a following PR.